src/services/codefixes/fixAddMissingMember.ts

```diff
@@ -80,37 +80,38 @@ namespace ts.codefix {
         },
     });
 
-    function getAllSuperClasses(cls: ClassLikeDeclaration | undefined, checker: TypeChecker): ReadonlyArray<ClassLikeDeclaration> {
+    function getAllSupers(decl: ClassOrInterface | undefined, checker: TypeChecker): ReadonlyArray<ClassOrInterface> {
         const res: ClassLikeDeclaration[] = [];
-        while (cls) {
-            const superElement = getClassExtendsHeritageElement(cls);
+        while (decl) {
+            const superElement = getClassExtendsHeritageElement(decl);
             const superSymbol = superElement && checker.getSymbolAtLocation(superElement.expression);
             const superDecl = superSymbol && find(superSymbol.declarations, isClassLike);
             if (superDecl) { res.push(superDecl); }
-            cls = superDecl;
+            decl = superDecl;
         }
         return res;
     }
 
+    type ClassOrInterface = ClassLikeDeclaration | InterfaceDeclaration;
     interface InfoBase {
         readonly kind: InfoKind;
         readonly token: Identifier;
-        readonly parentDeclaration: EnumDeclaration | ClassLikeDeclaration;
+        readonly parentDeclaration: EnumDeclaration | ClassOrInterface;
     }
     enum InfoKind { enum, class }

```

Comment:
Do you really need this enum since you can do this check through parentDeclaration.kind ?

Reply:
That currently wouldn't work to discriminate the union. See #18758.